Sources/NIOPosix/SocketChannel.swift

```diff
@@ -526,12 +526,24 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
     }
 
     override func connectSocket(to address: SocketAddress) throws -> Bool {
-        // For now we don't support operating in connected mode for datagram channels.
-        throw ChannelError.operationUnsupported
+        // TODO: this could be a channel option to do other things instead here, e.g. fail the connect
+        if !self.pendingWrites.isEmpty {
+            self.pendingWrites.failAll(
+                error: IOError(
```

Comment:
I think a NIO-specific error adds value because it helps us debug the exact cause of the problem. IMO, NIO shouldn't pretend that it isn't there when it is doing things that are unique and specific to it. In this case, we're making the decision to drop some data, and we should own that instead of pretending that the OS failed them.

Reply:
That reasoning makes total sense; I'll work on a more explicit error for this case where buffered writes are failed.